### PR TITLE
unzip: add -O/-I encoding support

### PR DESCRIPTION
**Pull request by Artoria2e5**
The synopsis line is quite questionable!

Please include this information, either here or in your commit messages for the
pull request, as appropriate (or delete the line if there's nothing relevant.

PR:	https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=271657.
Reported by:	Me
Reviewed by:	<If someone else reviewed your modification.>
Tested by:	<If someone else tested the change.>
Approved by:	<If you needed approval for this commit.>
MFC after:	<N [day[s]|week[s]|month[s]].  Request a reminder email>
Relnotes:	<Set to 'yes' for mention in release notes.>
Pull Request:	<https://github.com/freebsd/freebsd-src/pull/752>
Differential Revision:	<https://reviews.freebsd.org/D###>

### CODE REVIEW

**Comment by Artoria2e5** [comment]
that's a lot of CI for something that does not touch the kernel.

File: usr.bin/unzip/unzip.1 [hunk 1]
```diff
@@ -34,9 +34,10 @@
 .Sh SYNOPSIS
 .Nm
 .Op Fl aCcfjLlnopqtuvy
+.Op { Fl O | Fl I No } Ar encoding
 .Op Fl d Ar dir
 .Op Fl x Ar pattern
-.Op Fl P Ar password
+.Op Fl P Ar password
```
Comment (bsdimp) [inline usr.bin/unzip/unzip.1:40]: this looks like it adds trailing white space and should be removed.

File: usr.bin/unzip/unzip.1 [hunk 2]
```diff
@@ -62,6 +63,9 @@ Update existing.
 Extract only files from the zipfile if a file with the same name
 already exists on disk and is older than the former.
 Otherwise, the file is silently skipped.
+.It Fl I Ar encoding
+.It Fl O Ar encoding
```
Comment (bsdimp) [inline usr.bin/unzip/unzip.1:67]: Why both -I and -O?

**Comment by bsdimp** [comment]
> I hope your CI has build caches... This is a lot of world-building (haha, get it?) for one package changed.

Since you are a first-time submitter, I have to approve the CI workflow... which I've not done because of the issues I pointed out so far and it would be a waste.

File: usr.bin/unzip/unzip.1 (hunk 1 above)
Comment (Artoria2e5) [reply usr.bin/unzip/unzip.1:40]: Oops!

**Comment by Artoria2e5** [comment]
See Bugzilla. Both were in use for a long time. The info-zip design was to define "OEM" and "ISO" code pages with some FS defection to guess what system the zip came from (and thus whether to use iso/OEM), but that's a lot of work for us and not much gain.

**Comment by bsdimp** [comment]
That all makes sense. Pushed to main as a227908571d along with a1db51dbdb01 to fix a pre-existing igor complaint.

**Pull request closed without merging**